File: 2022/day3/part2.py
totalPriority = 0

with open("input.txt") as f:
  lines = f.readlines()
  for i in range(0, len(lines), 3):
    s = (set(lines[i]) & set(lines[i+1]) & set(lines[i+2])) - {"\n"}
    c = ord(s.pop())
    if 65 <= c <= 90:
      totalPriority += c - 38
    elif 97 <= c <= 122:
      totalPriority += c - 96

print(f"{totalPriority=}")

File: 2022/day3/part2.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
)

func main() {
	file, err := os.Open("input.txt")
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)

	totalPriority := 0

	for scanner.Scan() {
		rucksack1 := scanner.Text()
		scanner.Scan()
		rucksack2 := scanner.Text()
		scanner.Scan()
		rucksack3 := scanner.Text()
		elf1 := map[byte]bool{}
		elf2 := map[byte]bool{}
		elf3 := map[byte]bool{}
		for _, c := range []byte(rucksack1) {
			elf1[c] = true
		}
		for _, c := range []byte(rucksack2) {
			elf2[c] = true
		}
		for _, c := range []byte(rucksack3) {
			elf3[c] = true
		}
		// Find the intersection of the sets
		for k := range elf1 {
			if elf2[k] && elf3[k] {
				if k >= 65 && k <= 90 {
					totalPriority += int(k) - 38
				} else if k >= 97 && k <= 122 {
					totalPriority += int(k) - 96
				}
				break
			}
		}
	}

	fmt.Printf("Total priority: %d\n", totalPriority)

}
